File: 2_basic_knowledge/ruby/code_009.rb
n, s = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

answer = "No"
(0...(1 << n)).each do |i|
  partsum = 0
  (0...n).each do |j|
    if(i & (1 << j)) != 0
      partsum += a[j]
    end
  end

  if partsum == s
    answer = "Yes"
    break
  end
end

puts answer

File: 2_basic_knowledge/python/code_009.py
N, S = map(int, input().split)
A = list(map(int, input().split()))

answer = "No"
for i in range(0, 1 << N):
    partsum = 0
    for j in range(0, N):
        if(i & (1 << j)) != 0:
            partsum += A[j]
    if partsum == S:
        answer = "Yes"
        break

print(answer)
